etc/eslint/rules/typescript.js

```diff
@@ -0,0 +1,2526 @@
+
+/**
+* Disallows throwing literals as exceptions.
+*
+* @name no-throw-literal
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-throw-literal]{@link https://eslint.org/docs/rules/no-throw-literal}
+*/
+rules[ 'no-throw-literal' ] = 'error';
+
+/**
+* Disallows trailing whitespace at the end of lines.
+*
+* @name no-trailing-spaces
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-trailing-spaces]{@link https://eslint.org/docs/rules/no-trailing-spaces}
+*/
+rules[ 'no-trailing-spaces' ] = 'error';
+
+/**
+* Disallows initializing variables to `undefined`.
+*
+* @name no-undef-init
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-undef-init]{@link https://eslint.org/docs/rules/no-undef-init}
+*/
+rules[ 'no-undef-init' ] = 'error';
+
+/**
+* Disallows dangling underscores in identifiers.
+*
+* @name no-underscore-dangle
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-underscore-dangle]{@link https://eslint.org/docs/rules/no-underscore-dangle}
+*/
+rules[ 'no-underscore-dangle' ] = 'error';
+
+/**
+* Disallows control flow statements in `finally` blocks.
+*
+* @name no-unsafe-finally
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unsafe-finally]{@link https://eslint.org/docs/rules/no-unsafe-finally}
+*/
+rules[ 'no-unsafe-finally' ] = 'error';
+
+/**
+* Disables the rule against unused expressions.
+*
+* @name no-unused-expressions
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [no-unused-expressions]{@link https://eslint.org/docs/rules/no-unused-expressions}
+*/
+rules[ 'no-unused-expressions' ] = 'off';
+
+/**
+* Disallows unused labels.
+*
+* @name no-unused-labels
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unused-labels]{@link https://eslint.org/docs/rules/no-unused-labels}
+*/
+rules[ 'no-unused-labels' ] = 'error';
+
+/**
+* Disables the rule against unused variables.
+*
+* @name no-unused-vars
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [no-unused-vars]{@link https://eslint.org/docs/rules/no-unused-vars}
+*/
+rules[ 'no-unused-vars' ] = 'off';
+
+/**
+* Disables the rule against using variables before they are defined.
+*
+* @name no-use-before-define
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [no-use-before-define]{@link https://eslint.org/docs/rules/no-use-before-define}
+*/
+rules[ 'no-use-before-define' ] = 'off';
+
+/**
+* Disallows unnecessary constructors.
+*
+* @name no-useless-constructor
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-useless-constructor]{@link https://eslint.org/docs/rules/no-useless-constructor}
+*/
+rules[ 'no-useless-constructor' ] = 'error';
+
+/**
+* Disallows the use of the `void` operator.
+*
+* @name no-void
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-void]{@link https://eslint.org/docs/rules/no-void}
+*/
+rules[ 'no-void' ] = 'error';
```

Comment:
We use the `void` operator in JS. Is there a reason to disallow here?

Reply:
Changed it to allow usage of `void`.